libevmasm/Assembly.cpp

```diff
@@ -431,13 +431,16 @@ map<u256, u256> const& Assembly::optimiseInternal(
 	if (m_tagReplacements)
 		return *m_tagReplacements;
 
+	assertThrow(_settings.isCreation == m_creation, OptimizerException, "Mismatching creation settings.");
```

Comment:
We could also remove the flag in the settings entirely...

Reply:
Yeah, that actually sounds better.

Reply:
Done in another commit now.